src/final2/subtests/InvalidCommandTest.java

```diff
@@ -0,0 +1,915 @@
+package final2.subtests;
+
+import org.junit.Test;
+
+import test.Input;
+import test.runs.ErrorRun;
+import test.runs.NoOutputRun;
+import test.runs.Run;
+
+/**
+ * Tests the program for error messages with not well-formed or invalid commands.
+ * 
+ * @author Roman Langrehr
+ * @since 26.03.2015
+ * @version 1.0
+ *
+ */
+public class InvalidCommandTest extends LangtonSubtest {
+
+	private static final String[] ARGUMENT_COMMANDS = new String[] {
+			"create",
+			"field",
+			"position",
+			"direction",
+			"escape",
+			"move"
+	};
+
+	private static final String[] INVALID_ANT_NAMES = new String[] {
+			"ab",
+			"�",
+			"*"
+	};
+
+	private static final String[] INVALID_NUMBERS = new String[] {
+			"a",
+			"12t",
+			"three",
+			"12.5",
+			"$",
+			"9999999999999999999999999999999999999999999999999999999999999999"
+	};
+
+	private static final String[] NO_ARGUMENT_COMMANDS = new String[] {
+			"ant",
+			"print",
+			"quit"
+	};
+
+	private static final String[] OUT_OF_BOUNDS_COORDINATES_FOR_4X4 = new String[] {
+			"-1",
+			"5"
+	};
+
+	private static final String[] VALID_ARGUMENTS = new String[] {
+			"h,1,3",
+			"1,2",
+			"a",
+			"a",
+			"a",
+			"1"
+	};
+
+	/**
+	 * Asserts that the program avoids the creation of ants on blocked cells.
+	 */
+	@Test
+	public void createAntOnBlockedCellTest() {
+		inputFile = new String[] {
+				"0000",
+				"0*00",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("create h,1,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0b00",
+				"0a00",
+				"0000"
+		};
+		runs = new Run[] {
+				new ErrorRun("create h,1,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+
+		inputFile = new String[] {
+				"0000",
+				"0000",
+				"0A00",
+				"0000"
+		};
+		runs = new Run[] {
+				new NoOutputRun("move 1"),
+				new ErrorRun("create h,1,1"),
+				quit()
+		};
+		sessionTest(runs, Input.getFile(inputFile));
+	}

```

Comment:
> And I don't think that somebody checks for the case, that this cell's ant has the same name as the current ant and handle this special case separately.

Someone might try to create a "smart" and think this was not an error.
Or some silly things like somebody clones their field array, sets the new ant and then checks for duplicates.
It's certainly possible to ([un]intentionally) have program logic like that.